suitcase/Csv/__init__.py

```diff
@@ -0,0 +1,267 @@
+
+    >>> export(gen, '', '{time:%%Y-%%m-%%d_%%H:%%M}-')
+
+    Place the files in a different directory, such as on a mounted USB stick.
+
+    >>> export(gen, '/path/to/my_usb_stick')
+    """
+    serializer = Serializer(directory, file_prefix, **kwargs)
+    try:
+        for item in gen:
+            serializer(*item)
+    finally:
+        serializer.close()
+
+    return serializer.artifacts
+
+
+class Serializer(event_model.DocumentRouter):
+    """
+    Serialize a stream of documents to a set of csvs.
+
+    This creates a file named:
+    ``<directory>/<file_prefix>{stream_name}.csv``
+    for every Event stream that contains 1D 'tabular like' data.
+
+    .. warning::
+
+        This process explicitly ignores all data that is not 1D and does not
+        include any metadata in the output file.
+
+
+    .. note::
+
+        This can alternatively be used to write data to generic buffers rather
+        than creating files on disk. See the documentation for the
+        ``directory`` parameter below.
+
+    Parameters
+    ----------
+    directory : string, Path or Manager.
+        For basic uses, this should be the path to the output directory given
+        as a string or Path object. Use an empty string ``''`` to place files
+        in the current working directory.
+
+        In advanced applications, this may direct the serialized output to a
+        memory buffer, network socket, or other writable buffer. It should be
+        an instance of ``suitcase.utils.MemoryBufferManager`` and
+        ``suitcase.utils.MultiFileManager`` or any object implementing that
+        inferface. See the suitcase documentation (LINK ONCE WRITTEN) for
+        details.
+
+    file_prefix : str, optional
+        The first part of the filename of the generated output files. This
+        string may include templates as in ``{proposal_id}-{sample_name}-``,
+        which are populated from the RunStart document. The default value is
+        ``{uid}-`` which is guaranteed to be present and unique. A more
+        descriptive value depends on the application and is therefore left to
+        the user.
+
+    stack_images : Boolean
+        This indicates if we want one image per file (`stack_images` = `False`)
+        or many images per file (`stack_images` = `True`). If using
+        `stack_images` = `False` then an additional image number is added to
+        the file name.
+
+    **kwargs : kwargs
+        kwargs to be passed to ``pandas.Dataframe.to_csv``.
+    """
+    def __init__(self, directory, file_prefix='{uid}-', **kwargs):
+
+        if isinstance(directory, (str, Path)):
+            self.manager = suitcase.utils.MultiFileManager(directory)
+        else:
+            self.manager = directory
+
+        self.artifacts = self.manager._artifacts
+        self._streamnames = {}  # maps descriptor uids to stream_names
+        self._files = defaultdict(list)  # maps stream_name to file lists
```

Comment:
Something is out of sync here. Below this is treated as a dict of files objects, but here it's a `defaultdict` of lists. I think this should probably just be changed to `{}`.

Reply:
done